src/components/Button/Button.tsx

```diff
@@ -0,0 +1,74 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type {
+  HTMLAttributes,
+  ElementType,
+  MouseEventHandler,
+  ReactNode,
+} from "react";
+
+export type Props = {
+  appearance?: string;
+  children?: ReactNode;
+  className?: string;
+  dense?: boolean;
+  disabled?: boolean;
+  element?: ElementType;
+  hasIcon?: boolean;
+  inline?: boolean;
+  onClick?: (
+    evt: MouseEvent | React.MouseEvent | MouseEventHandler<HTMLButtonElement>
```

Comment:
I've updated this to `React.MouseEvent` as that seems to be the only type it will actually accept. I've also updated the Button type to extend the base element (`} & HTMLAttributes<HTMLElement>;`) as this component could be a `button`, `anchor` or any other element that needs to appear as a button.